### DISCUSSION TITLE
特定情况下，内存会飙升近十倍，且页面卡顿

### DISCUSSION THREAD

**Discussion by superchenwb** [alibaba/designable#34] [Q&A] [answered]
在webpack中的externals添加@formily/reactive，且启动Formily DevTools插件后，拖拽节点或者选择节点内存飙升近10倍
没拖拽一个节点到视图中，都会卡顿将近3秒
![image](https://user-images.githubusercontent.com/17635660/128294144-06c84d8d-3e7d-46e4-81f9-5b5c32aa5918.png)

如果不将@formily/reactive添加到externals，或者不启动Formily DevTools，都不会出现这个问题

**Comment by janryWang** [comment]
试试把右侧配置面板干掉，看看内存会不会溢出

**Reply by superchenwb** [reply]
试过，这样就不卡了

**Reply by superchenwb** [reply]
目前我把插件关掉，@formily/reactive添加到externals，内存占用感觉还是比官网的大一点，打开什么都不做，大概有50MB，后续会降到30多。官网的打开就是30MB，也更加稳定。

**Reply by janryWang** [reply]
如果只是externals加上，不启动devtools和没开启sandbox，也会内存飙升么

**Reply by superchenwb** [reply]
sandbox是否开启不影响，如果只是externals加上，不启动devtools，不会飙升

**Comment by janryWang** [comment]
复现了，但是我发现跟formily devtools没关系，是跟React Devtools有关系，隐身模式保留formily devtools是没内存泄漏问题的

**Reply by superchenwb** [reply]
但是我并没有安装React Devtools插件

**Reply by janryWang** [reply]
那我不清楚了，目前排查下来就跟ReactDevtools相关，我个人猜想是fiber node释放不了导致的泄漏

**Reply by superchenwb** [reply]
谢啦，我基本也没用过formily devtools

**Reply by janryWang** [reply]
你要不录个视频看看？

**Comment by janryWang** [comment]
麻烦录个视频看看，具体的复现路径

**Reply by superchenwb** [reply]
尝试过复现，居然没复现出现。
使用的formily项目进行测试，从最新的代码，一直回退到2.0.0.83版本都没复现出来

**Comment by janryWang** [comment, marked as answer]
ok，那就是没这个问题了
